proposals/lip-0051.md

```diff
@@ -814,484 +2596,147 @@ genesisTokenStoreSchema = {
     * All elements of the `lockedBalances` array must have `amount != 0`.
     * Either `availableBalance != 0` or `lockedBalances != []` must be true, i.e.: the entry must contain some tokens.
-  * Across all element of the `supplySubstore` array, all values given for `localID` must be unique.
-  * The `supplySubstore` array must be in ascending order of `localID`.
-  * Across all elements of the `escrowSubstore` array, the pairs `escrowChainID, localID` must be unique (each pair appears at most once).
-  * The `escrowSubstore` array must be in ascending order of `escrowChainID`. For a given `escrowChainID`, the entries must be in ascending order of `localID`.
-  * In the `terminatedEscrowSubstore` array, each element must be unique (appears at most once).
-  * The `terminatedEscrowSubstore` array must be in ascending order.
+  * Across all element of the `supplySubstore` array, all values given for `tokenID` must be unique.
+  * The `supplySubstore` array must be in ascending order of `tokenID`.
+  * Across all elements of the `escrowSubstore` array, the `tokenID` must be unique (it appears at most once).
+  * The `escrowSubstore` array must be in ascending order of `tokenID`. 
 
 * For each entry `userEntry` in `genesisBlockAssetObject.userSubstore`, create an entry in the user substore with:
-  ```python
-  storeKey = userEntry.address || uint32be(userEntry.tokenID.chainID) || uint32be(userEntry.tokenID.localID)
-  storeValue = {"availableBalance": userEntry.availableBalance,
-                "lockedBalances": [{"moduleID": lockedBalance.moduleID,
-                                    "amount": lockedBalance.amount
-                                   } for each lockedBalance in userEntry.lockedBalances]
-               } serialized using userStoreSchema.
-  ```
+```python
+storeKey = userEntry.address + userEntry.tokenID
+storeValue = encode(
+    schema=userStoreSchema,
+    object={
+        "availableBalance": userEntry.availableBalance,
+        "lockedBalances": [{
+            "module": lockedBalance.module,
+            "amount": lockedBalance.amount
+        } for each lockedBalance in userEntry.lockedBalances]
+)
+```
 
 * For each entry `supplyEntry` in `genesisBlockAssetObject.supplySubstore`, create an entry in the supply substore with:
-  ```python
-  storeKey = uint32be(supplyEntry.localID)
-  storeValue = {"totalSupply": supplyEntry.totalSupply} serialized using supplyStoreSchema.
-  ```
+```python
+storeKey = supplyEntry.tokenID
+storeValue = encode(
+    schema=supplyStoreSchema,
+    object={"totalSupply": supplyEntry.totalSupply}
+)
+```
 
 * For each entry `escrowEntry` in `genesisBlockAssetObject.escrowSubstore`, create an entry in the escrow substore with:
-  ```python
-  storeKey = uint32be(escrowEntry.escrowChainID) || uint32be(escrowEntry.localID)
-  storeValue = {"amount": escrowEntry.amount} serialized using escrowStoreSchema.
-  ```
+```python
+storeKey = escrowEntry.tokenID
```

Comment:
store key should be concat with escrowChainID?

Reply:
You are right. Added.